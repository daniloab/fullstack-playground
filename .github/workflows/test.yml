name: Tests

on:
  push:
    branches:
      - master
  pull_request:
    branches:
      - master

jobs:
  build:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v2
      - name: Setup Node.js
        uses: actions/setup-node@v2
        with:
          node-version: '14'
      - run: yarn
      - run: yarn jest