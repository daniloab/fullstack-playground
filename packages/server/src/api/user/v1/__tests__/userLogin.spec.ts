import { clearDbAndRestartCounters, connectMongoose, disconnectMongoose } from '../../../../../test';

import { createApiCall } from '../../../../../test/helper';
import { createUser } from '../../../../../test/createResource';

beforeAll(connectMongoose);

beforeEach(clearDbAndRestartCounters);

afterAll(disconnectMongoose);

const url = '/api/user/v1/login';

it('should not to do login if user not exists', async () => {
  const payload = {
    password: '123456',
    email: 'test@test.com',
  };

  const response = await createApiCall({ url, payload });

  expect(response.body).toMatchSnapshot();
  expect(response.body.message).toBe('Email or password incorrect');
  expect(response.status).toBe(400);
});

it('should not login if user exists but password is incorrect', async () => {
  await createUser({ email: 'test@test.com', password: '123456' });

  const payload = {
    password: 'abcdef',
    email: 'test@test.com',
  };

  const response = await createApiCall({ url, payload });

  expect(response.body).toMatchSnapshot();
  expect(response.body.message).toBe('Email or password incorrect');
  expect(response.status).toBe(400);
});

it('should made login if user exists and password is correct', async () => {
  await createUser({ email: 'test@test.com', password: '123456' });

  const payload = {
    password: '123456',
    email: 'test@test.com',
  };

  const response = await createApiCall({ url, payload });

  expect(response.body.message).toMatchSnapshot();
  expect(response.body.message).toBe('Login successful');
  expect(response.status).toBe(200);
});
