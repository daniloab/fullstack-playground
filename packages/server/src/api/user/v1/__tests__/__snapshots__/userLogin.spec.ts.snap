// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should made login if user exists and password is correct 1`] = `"Login successful"`;

exports[`should not login if user exists but password is incorrect 1`] = `
Object {
  "message": "Email or password incorrect",
  "status": "ERROR",
  "token": null,
}
`;

exports[`should not to do login if user not exists 1`] = `
Object {
  "message": "Email or password incorrect",
  "status": "ERROR",
  "user": null,
}
`;
